### PR TITLE
Fence position

### PR DESCRIPTION
**Pull request by cdb-is-not-good**
A new prospector. Shop bitch. code ready, too eepy to fiinish the rest

## About The Pull Request
<details>
<summary>
	About The Pull Request
</summary>
<hr>

	
<hr>
</details>

## Changelog
:cl:

/:cl:

### CODE REVIEW

File: code/__DEFINES/jobs.dm [hunk 1]
```diff
@@ -7,11 +7,11 @@
 #define JOBS_LSS "Chief Executive Officer","Cargo Technician","Lonestar Miner","Bartender","Chef","Gardener","Janitor","Artist"
 #define JOBS_CIVILIAN "Colonist", "Visitor"
 #define JOBS_CHURCH "Prime", "Vector"
-#define JOBS_PROSPECTOR "Foreman","Salvager","Prospector"
+#define JOBS_PROSPECTOR "Foreman","Dealer","Salvager","Prospector"
```
Comment (Trilbyspaceclone) [inline code/__DEFINES/jobs.dm:10]:
```suggestion
#define JOBS_PROSPECTOR "Foreman","Fence","Salvager","Prospector"
```

File: code/__DEFINES/jobs.dm [hunk 2]
```diff
@@ -7,11 +7,11 @@
...
 #define JOBS_NONHUMAN "AI","Robot","pAI"
 #define JOBS_INDEPENDENT "Lodge Hunt Master","Lodge Hunter","Lodge Herbalist","Outsider"
 
-#define JOBS_ANTI_HIVEMIND "Blackshield Commander","Warrant Officer","Supply Specialist","Ranger","Corpsman","Blackshield Trooper","Marshal Officer","Sergeant","Prime","Vector","Foreman","Salvager","Prospector","Premier","Steward","AI","Janitor","Soteria Lifeline Technician","Soteria Roboticist","Lonestar Miner"
+#define JOBS_ANTI_HIVEMIND "Blackshield Commander","Warrant Officer","Supply Specialist","Ranger","Corpsman","Blackshield Trooper","Marshal Officer","Sergeant","Prime","Vector","Foreman","Dealer","Salvager","Prospector","Premier","Steward","AI","Janitor","Soteria Lifeline Technician","Soteria Roboticist","Lonestar Miner"
```
Comment (Trilbyspaceclone) [inline code/__DEFINES/jobs.dm:14]:
```suggestion
#define JOBS_ANTI_HIVEMIND "Blackshield Commander","Warrant Officer","Supply Specialist","Ranger","Corpsman","Blackshield Trooper","Marshal Officer","Sergeant","Prime","Vector","Foreman","Fence","Salvager","Prospector","Premier","Steward","AI","Janitor","Soteria Lifeline Technician","Soteria Roboticist","Lonestar Miner"
```

File: code/datums/outfits/jobs/prospector.dm [hunk 3]
```diff
@@ -7,6 +7,14 @@
 	id_type = /obj/item/card/id/black
 	pda_type = /obj/item/modular_computer/pda/cargo
 
+/decl/hierarchy/outfit/job/dealer
```
Comment (Trilbyspaceclone) [inline code/datums/outfits/jobs/prospector.dm:10]:
```suggestion
/decl/hierarchy/outfit/job/fence
```

File: code/datums/outfits/jobs/prospector.dm [hunk 4]
```diff
@@ -7,6 +7,14 @@
...
+	name = OUTFIT_JOB_NAME("Prospectors - Dealer")
```
Comment (Trilbyspaceclone) [inline code/datums/outfits/jobs/prospector.dm:11]:
```suggestion
	name = OUTFIT_JOB_NAME("Prospectors - Fence")
```

File: code/game/jobs/job/prospector.dm [hunk 5]
```diff
@@ -55,6 +55,58 @@
...
+	selection_color = "#97b0be"
+	req_admin_notify = 1
+	wage = WAGE_LABOUR_HAZARD
+	department_account_access = TRUE
+	health_modifier = 5
+
+	disallow_species = list(FORM_SOTSYNTH, FORM_AGSYNTH, FORM_BSSYNTH, FORM_CHURCHSYNTH, FORM_NASHEF)
+
+
+	outfit_type = /decl/hierarchy/outfit/job/dealer
```
Comment (Trilbyspaceclone) [inline code/game/jobs/job/prospector.dm:77]:
```suggestion
	outfit_type = /decl/hierarchy/outfit/job/fence
```

File: code/game/objects/structures/crates_lockers/closets/secure/cargo.dm [hunk 6]
```diff
@@ -642,6 +642,131 @@
 		new /obj/item/grenade/spawnergrenade/manhacks/junkbot(src)
 
 
+/obj/structure/closet/secure_closet/reinforced/foreman/dealer  //essentially just a modified foreman locker. This is fine for now.
```
Comment (Trilbyspaceclone) [inline code/game/objects/structures/crates_lockers/closets/secure/cargo.dm:645]:
```suggestion
/obj/structure/closet/secure_closet/reinforced/foreman/fence //essentially just a modified foreman locker. This is fine for now.
```

File: code/game/objects/structures/crates_lockers/closets/secure/cargo.dm [hunk 7]
```diff
@@ -642,6 +642,131 @@
...
+	name = "Fences locker"
+	req_access = list(access_fence)
+	icon_state = "dealer"
```
Comment (Trilbyspaceclone) [inline code/game/objects/structures/crates_lockers/closets/secure/cargo.dm:648]:
```suggestion
	icon_state = "fence"
```
Going to have to change the iconstate in dmi too

File: code/game/objects/structures/crates_lockers/closets/secure/cargo.dm [hunk 8]
```diff
@@ -642,6 +642,131 @@
...
+
+//Small note from Trilby: Forman loot picks are basically just the BEST versons of the scav lockers so theirs no doble coin flip
+//Also dosnt have a side arm loot spawn but still has the vars for one if someone wants to mirrow it or do something like that in furchture
+/obj/structure/closet/secure_closet/reinforced/foreman/dealer/gain_rng()
```
Comment (Trilbyspaceclone) [inline code/game/objects/structures/crates_lockers/closets/secure/cargo.dm:652]:
```suggestion
/obj/structure/closet/secure_closet/reinforced/foreman/fence/gain_rng()
```

File: code/game/objects/structures/crates_lockers/closets/secure/cargo.dm [hunk 9]
```diff
@@ -642,6 +642,131 @@
...
+	main_weapon_cache = pickweight(list("SMG" = 12, "LMG" = 4, "GOOD_SMG" = 10, "SHOTGUN" = 8, "RIFLE" = 6, "EGUN" = 4))
+	side_arm_cache = pickweight(list("PISTOL" = 12, "REVOLVER" = 4, "GOOD_PISTOL" = 10, "HEAVY_PISTOL" = 2, "EGUN_P" = 4))
+	melee_cache = pickweight(list("MACHETE" = 12, "KATANA" = 8, "PHAMMER" = 4, "CHAINSAW" = 1))
+	bag_cache = pickweight(list("INDUSTRIAL" = 24, "DUFFEL" = 16, "MILI" = 8, "BLUESPACE" = 1))
+	armor_cache = pickweight(list("BASIC_A" = 12, "BULLET_A" = 4, "EGUN_A" = 4, "MELEE_A" = 8))
+
+/obj/structure/closet/secure_closet/reinforced/foreman/dealer/populate_contents()
```
Comment (Trilbyspaceclone) [inline code/game/objects/structures/crates_lockers/closets/secure/cargo.dm:659]:
```suggestion
/obj/structure/closet/secure_closet/reinforced/foreman/fence/populate_contents()
```

File: maps/__Nadezhda/map/_Nadezhda_Colony_New.dmm [hunk 10]
```diff
@@ -109595,12 +109602,11 @@
 /turf/simulated/floor/reinforced,
 /area/nadezhda/engineering/atmos)
 "wSC" = (
-/obj/structure/table/rack,
-/obj/effect/floor_decal/industrial/warningwhite/full,
 /obj/structure/sign/warning/smoking/small{
 	pixel_x = 32;
 	pixel_y = -32
 	},
+/obj/structure/closet/secure_closet/reinforced/foreman/dealer,
```
Comment (Trilbyspaceclone) [inline maps/__Nadezhda/map/_Nadezhda_Colony_New.dmm:109609]:
```suggestion
/obj/structure/closet/secure_closet/reinforced/foreman/fence,
```

**Review by Trilbyspaceclone — CHANGES_REQUESTED** [review]
Fence > Dealer for code back end consectiy

**Comment by Trilbyspaceclone** [comment]
The icon in the dmi needs to be fixed, other then that it i think you just need to test if it works

File: code/game/jobs/department.dm [hunk 11]
```diff
@@ -126,7 +126,7 @@
 	id = DEPARTMENT_PROSPECTOR
 	//With only the Foreman currently being paid, after 8 hours, it totals to 4800 of payment, leaving an ample 1700 left.
 	account_initial_balance = 6500 //With how Prospectors no longer get paid, they no longer need such an inflated department balance
-	jobs_in_department = list("/datum/job/foreman","/datum/job/salvager","/datum/job/pro")
+	jobs_in_department = list("/datum/job/foreman","/datum/job/salvager","/datum/job/pro,datm/job/fence")
```
Comment (Trilbyspaceclone) [inline code/game/jobs/department.dm:129]:
```suggestion
	jobs_in_department = list("/datum/job/foreman","/datum/job/salvager","/datum/job/pro","/datm/job/fence")
```

**Comment by cdb-is-not-good** [comment]
> The icon in the dmi needs to be fixed, other then that it i think you just need to test if it works

which icon? it was working last i tested iit

**Pull request merged**